### PR TITLE
Added Greedy and Divide&conquer Algo

### PR DESCRIPTION
**Pull request by ankitvashisht12**
Added greedy algorithm and Divide_and_conquer Algorithms

### CODE REVIEW

File: Divide_and_Conquer/binary_search.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    n = n_array[0]
+    array = n_array[1:]
+    k = (k_elements[0])
+    elements = k_elements[1:]
+    for i in range(k):
+        print(binary_search(array, elements[i]), end = ' ')
+    print()
+
+
+main()
```
Comment (vanshikaarora) [inline Divide_and_Conquer/binary_search.py]: Hey this code is showing compilation error https://ide.codingblocks.com/s/141187 with python
and runtime error with python3
Can you please check.
Thanks

Comment (ankitvashisht12) [reply Divide_and_Conquer/binary_search.py]: Fixed the source code 
Please review again. Thanks :)

File: Divide_and_Conquer/maximum-subarray.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+# Constraints. 1 ≤ m ≤ 10**3 .
+
+# Output Format. Output the starting index, ending index and value of sum of subarray.
+
+# Example :
+# -3 -8 6 -2 -3 1 5 -6                       <--- input
+#  2 6 7                                     <--- output
+
+import math as m
```
Comment (vanshikaarora) [inline Divide_and_Conquer/maximum-subarray.py]: I checked this file https://ide.codingblocks.com/s/141194.
It shows runtime error. Can you please check:)

Comment (ankitvashisht12) [reply Divide_and_Conquer/maximum-subarray.py]: The error is coming due to space after the input. I have now fixed this bug.
If it is still giving error, make sure to change the language from Python to Python3.
You can check out here too https://ide.geeksforgeeks.org/I8pm7SjKF0
Thanks :)

File: Divide_and_Conquer/maximum-subarray.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+            return (right_low, right_high, right_sum )
+        else:
+            return (cross_low, cross_high, cross_sum)
+
+def main():
+    array = [ int(i) for i in input().strip().split(" ")]
+    low, high, s = find_max_subarray( array, 0, len(array)-1)
+    print(low, high, s)
+
+main()
```
Comment (vanshikaarora) [inline Divide_and_Conquer/maximum-subarray.py]: Hey there is some runtime error in this file. Can you please check https://ide.codingblocks.com/s/141649

File: Divide_and_Conquer/maximum-subarray.py (hunk 2 above)
Comment (vanshikaarora) [reply Divide_and_Conquer/maximum-subarray.py]: Yeah Thanks for the changes

**Pull request merged**